libpod/container_internal.go

```diff
@@ -190,13 +191,28 @@ func (c *Container) setupStorage(ctx context.Context) error {
 		return errors.Wrapf(ErrInvalidArg, "must provide image ID and image name to use an image")
 	}
 
-	containerInfo, err := c.runtime.storageService.CreateContainerStorage(ctx, c.runtime.imageContext, c.config.RootfsImageName, c.config.RootfsImageID, c.config.Name, c.config.ID, c.config.MountLabel)
+	options := storage.ContainerOptions{IDMappingOptions: c.config.IDMappings}
+	containerInfo, err := c.runtime.storageService.CreateContainerStorage(ctx, c.runtime.imageContext, c.config.RootfsImageName, c.config.RootfsImageID, c.config.Name, c.config.ID, c.config.MountLabel, &options)
 	if err != nil {
 		return errors.Wrapf(err, "error creating container storage")
 	}
 
+	if len(c.config.IDMappings.UIDMap) != 0 || len(c.config.IDMappings.GIDMap) != 0 {
+		c.state.UserNSRoot, err = ioutil.TempDir("", fmt.Sprintf("libpod-%s", c.ID()))
```

Comment:
Can we put this tempdir in libpod's temporary dir?

Reply:
unfortunately `/var/run/containers/storage` is owned by root and has mode 0700 so other users cannot access directories there

Reply:
`libpod` defaults to `/var/lib/libpod` (https://github.com/projectatomic/libpod/blob/master/libpod/runtime.go#L164). I'm fine with changing the permissions there, so long as we make sure the database and a few other files are 0600

Reply:
Why put temporary data in a directory other then /run?

Reply:
Actually it's `/var/run/libpod`

I blame lack of caffeine

Reply:
hmm.. `containerInfo.RunDir` here is under `/var/run/containers/storage/overlay-containers/`, I've added a debug Printf:
```
diff --git a/libpod/container_internal.go b/libpod/container_internal.go
index e560caf..08752a1 100644
--- a/libpod/container_internal.go
+++ b/libpod/container_internal.go
@@ -210,6 +210,7 @@ func (c *Container) setupStorage(ctx context.Context) error {
        c.config.StaticDir = containerInfo.Dir
        c.state.RunDir = containerInfo.RunDir
        c.state.DestinationRunDir = c.state.RunDir
+       fmt.Printf("containerInfo.RunDir=%s\n", containerInfo.RunDir)
        if c.state.UserNSRoot != "" {
                c.state.DestinationRunDir = filepath.Join(c.state.UserNSRoot, "rundir")
        }
```

 and I get:

```
sudo bin/podman run --rm --uidmap=0:1:70000 --gidmap=0:20000:70000 alpine true
containerInfo.RunDir=/var/run/containers/storage/overlay-containers/13faaf7658362f4473eaf6e2fa3ce16059e21b8a0ecbe12eac96279c9ed6063e/userdata
```

Reply:
Oops, should have been more clear - it's a separate directory... `c.runtime.config.TmpDir` is how you'd usually get at it. It's separate from c/storage's temporary directory, so we can be a bit more liberal with permissions.

Reply:
yes sure, that makes sense.  I've just updated it in the new fixup

It took a while to debug an issue with runc, and again it was that it doesn't like symlinks :(